rcsb/workflow/bcif/task_functions.py

```diff
@@ -0,0 +1,414 @@
+    counter=[0],
+):
+    if contentType == "pdb":
+        pdbId = pdbId.lower()
+    elif contentType == "csm":
+        pdbId = pdbId.upper()
+    remoteFileName = "%s%s" % (
+        pdbId,
+        outfileSuffix.replace(".bcif.gz", ".cif.gz").replace(".bcif", ".cif"),
+    )
+    # copy or download to cifFilePath
+    if not remotePath.startswith("http"):
+        # local file
+        localFilePath = os.path.join(remotePath, remoteFileName)
+        if not os.path.exists(localFilePath):
+            logger.error("%s not found", localFilePath)
+            return
+        cifFilePath = os.path.join(tempPath, remoteFileName)
+        shutil.copy(localFilePath, cifFilePath)
+    else:
+        url = os.path.join(remotePath, pdbId[-3:-1], remoteFileName)
+        cifFilePath = os.path.join(tempPath, remoteFileName)
+        if contentType == "csm":
+            url = os.path.join(
+                remotePath, pdbId[0:2], pdbId[-6:-4], pdbId[-4:-2], remoteFileName
+            )
+            cifFilePath = os.path.join(tempPath, remoteFileName)
+        try:
+            r = requests.get(url, timeout=300, stream=True)
+        except requests.exceptions.ConnectTimeout as e:
+            logger.exception(str(e))
+            return
+        try:
+            if r and r.status_code < 400:
+                dirs = os.path.dirname(cifFilePath)
+                if not os.path.exists(dirs):
+                    os.makedirs(dirs)
+                    try:
+                        os.chmod(dirs, 0o777)
+                        shutil.chown(dirs, "ubuntu", "ubuntu")
+                    except PermissionError as e:
+                        pass
+                with open(cifFilePath, "ab") as w:
+                    for chunk in r.raw.stream(1024, decode_content=False):
+                        if chunk:
+                            w.write(chunk)
+                try:
+                    shutil.chown(cifFilePath, "ubuntu", "ubuntu")
+                    os.chmod(cifFilePath, 0o777)
+                except PermissionError as e:
+                    pass
+            else:
+                raise requests.exceptions.RequestException(
+                    "error - request failed for %s" % url
+                )
+        except Exception as e:
+            logger.exception(str(e))
+            if os.path.exists(cifFilePath):
+                os.unlink(cifFilePath)
+            return
+
+    # form output path bcifFilePath
+    bcifFilePath = getBcifFilePath(
+        pdbId, outfileSuffix, updateBase, contentType, outputContentType, outputHash
+    )
+    if not bcifFilePath:
+        raise ValueError("failed to form bcif file path")
+    if os.path.exists(bcifFilePath):
+        # earlier timestamp ... overwrite
+        try:
+            os.unlink(bcifFilePath)
+            if os.path.exists(bcifFilePath):
+                raise Exception("file %s not removed" % bcifFilePath)
+        except Exception as e:
+            logger.exception(str(e))
```

Comment:
One can discuss if failing to remove a file is a critical error or not. I'm going to say it is important enough for the workflow to fail. So my recommendation here is to remove the try/catch block so that the exception is seen by the workflow engine.